From 3dbe90ca85ecaad15b58f84aea69f5459e41dda6 Mon Sep 17 00:00:00 2001
From: quintus-lab<noreply@github.com>
Date: Thu, 17 Jun 2021 12:44:33 +0800
Subject: [PATCH] dnsmasq add filter aaaa option

	modified:   package/base-files/files/etc/init.d/boot
	modified:   package/network/services/dnsmasq/files/dhcp.conf
	modified:   package/network/services/dnsmasq/files/dnsmasq.init
---
 package/base-files/files/etc/init.d/boot      |   1 +
 .../network/services/dnsmasq/files/dhcp.conf  |   1 +
 .../services/dnsmasq/files/dnsmasq.init       |   2 +
 .../dnsmasq/patches/910-mini-ttl.patch        | 102 ++++++++++++++++++
 .../patches/911-dnsmasq-filter-aaaa.patch     |  68 ++++++++++++
 5 files changed, 174 insertions(+)

diff --git a/package/base-files/files/etc/init.d/boot b/package/base-files/files/etc/init.d/boot
index a1e8e828dd..974b68860f 100755
--- a/package/base-files/files/etc/init.d/boot
+++ b/package/base-files/files/etc/init.d/boot
@@ -33,6 +33,7 @@ boot() {
 	mkdir -p /tmp/resolv.conf.d
 	touch /tmp/resolv.conf.d/resolv.conf.auto
 	ln -sf /tmp/resolv.conf.d/resolv.conf.auto /tmp/resolv.conf
+	ln -sf /tmp/resolv.conf.d/resolv.conf.auto /tmp/resolv.conf.auto
 	grep -q debugfs /proc/filesystems && /bin/mount -o noatime -t debugfs debugfs /sys/kernel/debug
 	grep -q bpf /proc/filesystems && /bin/mount -o nosuid,nodev,noexec,noatime,mode=0700 -t bpf bpffs /sys/fs/bpf
 	grep -q pstore /proc/filesystems && /bin/mount -o noatime -t pstore pstore /sys/fs/pstore
diff --git a/package/network/services/dnsmasq/files/dhcp.conf b/package/network/services/dnsmasq/files/dhcp.conf
index 8c42ef782e..5ff9bf8bfd 100644
--- a/package/network/services/dnsmasq/files/dhcp.conf
+++ b/package/network/services/dnsmasq/files/dhcp.conf
@@ -20,6 +20,7 @@ config dnsmasq
 	#list notinterface	lo
 	#list bogusnxdomain     '64.94.110.11'
 	option localservice	1  # disable to allow DNS requests from non-local subnets
+	option filteraaaa	1
 	option ednspacket_max	1232
 
 config dhcp lan
diff --git a/package/network/services/dnsmasq/files/dnsmasq.init b/package/network/services/dnsmasq/files/dnsmasq.init
index 680e72f9e7..d805fb3bd4 100644
--- a/package/network/services/dnsmasq/files/dnsmasq.init
+++ b/package/network/services/dnsmasq/files/dnsmasq.init
@@ -893,6 +893,8 @@ dnsmasq_start()
 	append_bool "$cfg" sequential_ip "--dhcp-sequential-ip"
 	append_bool "$cfg" allservers "--all-servers"
 	append_bool "$cfg" noping "--no-ping"
+	append_bool "$cfg" filteraaaa "--filter-aaaa"
+   append_parm "$cfg" mini_ttl "--min-ttl"
 	append_bool "$cfg" rapidcommit "--dhcp-rapid-commit"
 	append_bool "$cfg" scriptarp "--script-arp"
 
-- 
2.25.1

